Sources/Support/PaywallExtensions.swift

```diff
@@ -0,0 +1,165 @@
+            ids: offering.allProductIdentifiers,
+            prefersPromotionalIcon: prefersPromotionalIcon,
+            icon: icon,
+            placeholderIcon: placeholderIcon
+        )
+    }
+
+}
+
+@available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *)
+extension SubscriptionStoreView {
+
+    /// Creates a ``SubscriptionStoreView`` from an ``Offering``
+    /// with custom marketing content.
+    /// When the user purchases products through this paywall, the `RevenueCat` SDK will handle
+    /// the result automatically. All you need to do is to dismiss the paywall
+    ///
+    /// - Seealso: ``CurrentOfferingSubscriptionStoreView``
+    public init(
+        offering: Offering,
+        @ViewBuilder marketingContent: () -> (Content)
+    ) {
+        self.init(
+            productIDs: offering.subscriptionProductIdentifiers,
+            marketingContent: marketingContent
+        )
+    }
+
+    /// Creates a ``SubscriptionStoreView`` from an ``Offering``
+    /// that doesn't take a custom view to use for marketing content.
+    ///
+    /// - Seealso: ``CurrentOfferingSubscriptionStoreView``
+    public init(
+        offering: Offering
+    ) where Content == AutomaticSubscriptionStoreMarketingContent {
+        self.init(productIDs: offering.subscriptionProductIdentifiers)
+    }
+
+}
+
+/// ``_StoreKit_SwiftUI/SubscriptionStoreView`` that displays subscription products in the current ``Offering``.
+@available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *)
+public struct CurrentOfferingSubscriptionStoreView<Content: View>: View {
+
+    @State
+    private var currentOffering: Offering?
+    private let marketingContent: (() -> Content)?
+
+    /// Creates a view to load all subscriptions in a subscription group from the App Store.
+    ///
+    /// When the user purchases products through this paywall, the `RevenueCat` SDK will handle
+    /// the result automatically. All you need to do is to dismiss the paywall
+    public init() where Content == AutomaticSubscriptionStoreMarketingContent {
+        self.marketingContent = nil
+    }
+
+    /// Creates a view to load all subscriptions in a subscription group from the App Store, and merchandise
+    /// them with a custom marketing content.
+    ///
+    /// When the user purchases products through this paywall, the `RevenueCat` SDK will handle
+    /// the result automatically. All you need to do is to dismiss the paywall
+    public init(@ViewBuilder marketingContent: @escaping () -> Content) {
+        self.marketingContent = marketingContent
+    }
+
+    // swiftlint:disable:next missing_docs
+    public var body: some View {
+        Group {
+            if let currentOffering {
+                if let marketingContent {
+                    SubscriptionStoreView(offering: currentOffering,
+                                          marketingContent: marketingContent)
+                } else {
+                    SubscriptionStoreView(offering: currentOffering)
```

Comment:
A couple more questions about these views:
- When purchasing through this view, does it automatically finish the transaction? Or does the developer still have to do it?
- If the user purchases an offering through this view, we won't be getting the `presentedOfferingIdentifier` setup in the SDK... We need to figure out a way to set that.

Reply:
Will add to this, that anything we do in the `purchasePackage/Product` methods won't happen when they use this view. We should check if there are any other unintended side effects of this. Another one I see is that we would not be calling `preventPurchasePopupCallFromTriggeringCacheRefresh` so we would never set the offerings cache as stale I think.

Reply:
> When purchasing through this view, does it automatically finish the transaction? Or does the developer still have to do it?

Our SDK handles that :) (though we do need SDK-3170)

> If the user purchases an offering through this view, we won't be getting the presentedOfferingIdentifier setup in the SDK... We need to figure out a way to set that.

Oh great call. Just filed SDK-3176 for that.

Reply:
> Will add to this, that anything we do in the purchasePackage/Product methods won't happen when they use this view. We should check if there are any other unintended side effects of this.

I think that's fine. This is basically like our SDK acting in observer mode for a purchase being done from outside.

> Another one I see is that we would not be calling `preventPurchasePopupCallFromTriggeringCacheRefresh` so we would never set the offerings cache as stale I think.

Another reason to do SDK-3172 and get rid of that method.